src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -435,47 +491,76 @@ public static bool EncodeStringLiterals(ReadOnlySpan<string> values, string sepa
                 return EncodeStringLiteral(values[0], destination, out bytesWritten);
             }
 
-            if (destination.Length != 0)
+            if (destination.Length == 0)
             {
-                int valueLength = 0;
+                return false;
+            }
+
+            int valueLength = 0;
 
-                // Calculate length of all parts and separators.
-                foreach (string part in values)
+            // Calculate length of all parts and separators.
+            foreach (string part in values)
+            {
+                valueLength = checked(valueLength + part.Length);
+            }
+
+            valueLength = checked(valueLength + (values.Length - 1) * separator.Length);
+
+            int expectedHuffmanLength = Huffman.GetEncodedLength(values, separator, lowerCase: false);
+            int integerLength;
+
+            if (expectedHuffmanLength < valueLength)
```

Comment:
Wouldn't we want it to be significantly less in order to be worthwhile rather than just less?

Reply:
I'm curious, what would you consider "significantly less"?

The smallest Huffman encoding is 5 bits, so you will never get better than a 37.5% reduction. But only ~10 chars are encoded using 5 bits, and quite a few relatively common chars are encoded in 7 or more bits.

Just for arguments sake, let's assume the typical case is 25-30%. Is that worth it? Where would you draw the line?

Reply:
> Is that worth it? Where would you draw the line?

My assumption is "no", though at the same time obviously educated people felt there was worth in including huffman coding in the standard, so one would hope there's measurable value somewhere.

I would want to draw the line at where we actually see a benefit in doing this.  We're spending time to calculate the size, and then if using the coding would save even one byte, we're spending time again to do the compression and forcing the other side to spend time decompressing it.  I have trouble believing one byte is worth it.

Hence my earlier question:
"Is there any case where you've found it to make an impactful difference? Unless there's a really good performance case for it, I'd rather just leave it unimplemented and have simpler / less code."

Reply:
> We're spending time to calculate the size, and then if using the coding would save even one byte, we're spending time again to do the compression and forcing the other side to spend time decompressing it.

I wrote another implementation that optimistically encodes in a single pass. It was not measurably faster, and even more complex, so not bothering with it.